### ISSUE TITLE
Walking + Firing = Self Shooting

### ISSUE THREAD

**Issue by 9joshua** [ClobberXD/gunslinger#37]
If you walk forward and fire a weapon at the same time you shoot yourself. Is it possible to ignore the shooter when checking for hit objects or can the ray begin a bit further in front of the shooter?...

![opt](https://user-images.githubusercontent.com/20234985/84754817-01401480-b015-11ea-8270-2d8013599dff.gif)

**Comment by ClobberXD** [comment]
https://github.com/ClobberXD/gunslinger/blob/d20130a2db218825e419f8c3433fb5d04c83ee21/api.lua#L61-L67

The code already avoids the first pointed thing if the ray originates from within the pointed thing. Might it be possible that the ray encounters the same object again after skipping? That's highly unlikely...

Are you using the latest version ([v0.24-pre](https://github.com/ClobberXD/gunslinger/releases/tag/v0.24-pre)) of Gunslinger?

**Comment by 9joshua** [comment]
I cloned from the master branch 3 days ago. My version contains the code you pasted above.
It only happens when walking forward so maybe due to lag from the time the check is done to when it continues to trace?

**Comment by ClobberXD** [comment]
From the API docs:

> The map is loaded as the ray advances. If the map is modified after the
> `Raycast` is created, the changes may or may not have an effect on the object.

This means that unless the Raycast object is re-created, it shouldn't detect the same player again and again. But there is a possibility of the ray originating from outside of the shooter's collision box due to lag (is this what you were saying?), which means the code linked in the previous post will not detect the player properly. Ideally, it's better to explicitly check for the player; I wanted to avoid string comparisons as they're computationally expensive. I'll see if there's a better way to check if an object corresponds to the shooter themselves.

**Comment by 9joshua** [comment]
Yes, I think it could be due to lag from the time the ray is initiated to when the player moves forward and the ray progresses.
